src/__tests__/test_helpers/mockXHR.js

```diff
@@ -19,6 +19,9 @@ function mockXHR(results, options = {}) {
     getRequestHeader: function (key) {
       return headers[key];
     },
+    getAllResponseHeaders: function() {
```

Comment:
Why does this need to be added to the tests?

Reply:
There were about ~40 tests that failed with this error:

`XMLHttpRequest failed: "TypeError: xhr.getAllResponseHeaders is not a function"`

A lot of the tests uses the `RESTController` it looks like with a mocked out `xhr` object (some uses the mockXHR, a lot of others uses their own, could probably be cleaned up) and when it reaches the header check they failed.

Reply:
But the tests already passed previously with that mocked controller, right?

Reply:
If I take those lines out in this PR, we get the following result:
```
Test Suites: 3 failed, 46 passed, 49 total
Tests:       24 failed, 1047 passed, 1071 total
Snapshots:   0 total
Time:        42.54 s
```

Because `RestController.ts:dispatch()` in this PR needs `getAllResponseHeaders()` to make sure the header exists before calling `getRequestHeader`, that's how it avoids the Chrome error log (like this article suggests https://trackjs.com/blog/refused-unsafe-header/)

Before I did anything, the tests also passed yes if that's what you were asking?

Reply:
Yes, that's what I'm asking - why do we have to modify the tests, if they passed previously?

Reply:
RESTController now requires it. but if we do the other suggested change I guess these test changes aren't needed.

Reply:
We reverted this to not require it anymore.